### PR TITLE
Fix for enum field on SQLite database

### PR DESCRIPTION
**Pull request by promatik**
Reported on https://github.com/Laravel-Backpack/NewsCRUD/issues/55.

`enum` field doesn't work on SQLite database, Laravel changes it to a `varchar`, so in that case, it may take `options` for the html select.

### CODE REVIEW

**Comment by tabacitu** [comment]
Thank you @promatik !

The `enum` field will only work on MySQL afaik. So hard-coding the sqlite scenario isn't a "proper fix", a better conditional would be "if not MySQL" I think. 

But I guess the question here is... when someone is trying to use an `enum` field on a non-supported DMBS, do we want it to fail _silently_ (like this PR would do) or do we want it to fail _hard_ (like it was before). And I think failing hard is better - it would at least give an indication to the dev of what is going wrong. Otherwise, their experience would be "_I'm using the `enum` field, why doesn't it pick up the options_", which would be frustrating.

So I vote against this PR. What do you think about my reasoning above @promatik ?

**Review by pxpm — COMMENTED** [review]
Hey @promatik 

What you think of instead checking for SQLite, we use a `try => catch` block that if it fails (be it sqlite or whatever other dbms people are using that does not support enum)we just return the `null` ? 

Let me know, I think it would be a broader a better solution. 

EDIT AFTER READING @tabacitu reply:

- yeah .. probably better to just fail hard, but we can properly give an exception instead of complaining of the `QLSTATE[HY000]: General error: 1 near "SHOW": syntax error`

So I think we still can use the `try => catch` block, but instead of returning null, we will properly throw an exception that cleary states that `Enum field type is not supported.` 

Would this be the best of both worlds ? 


Best,
Pedro

**Comment by promatik** [comment]
@pxpm I think I agree, so, first I agree with @tabacitu, it may fail "hard", but maybe a `Enum field type is not supported.` is better than the current error.

File: src/app/Models/Traits/HasEnumFields.php [hunk 1]
```diff
@@ -20,7 +20,13 @@ public static function getPossibleEnumValues($field_name)
 
         $instance = new static(); // create an instance of the model to be able to get the table name
         $connectionName = $instance->getConnectionName();
-        $type = DB::connection($connectionName)->select(DB::raw('SHOW COLUMNS FROM `'.$table_prefix.$instance->getTable().'` WHERE Field = "'.$field_name.'"'))[0]->Type;
+
+        try {
+            $type = DB::connection($connectionName)->select(DB::raw('SHOW COLUMNS FROM `'.$table_prefix.$instance->getTable().'` WHERE Field = "'.$field_name.'"'))[0]->Type;
+        } catch (\Exception $e) {
+            abort(500, 'Enum field type is not supported.');
```
Comment (tabacitu) [inline src/app/Models/Traits/HasEnumFields.php:27]:
```suggestion
            abort(500, 'Enum field type is not supported - it only works on MySQL.');
```

**Pull request merged**